packages/ui-components/src/ExpandableErrorDialog.tsx

```diff
@@ -31,58 +31,53 @@ interface IErrorDialogProps {
   default_msg: string;
 }
 
-export class ExpandableErrorDialog extends React.Component<
-  IErrorDialogProps,
-  any
-> {
-  dialogNode: HTMLDivElement;
-  collapsedDimensions: number[];
+export const ExpandableErrorDialog: React.FC<IErrorDialogProps> = ({
+  reason,
+  message,
+  timestamp,
+  traceback,
+  default_msg
+}) => {
+  const [collapsedSize, setCollapsedSize] = React.useState(null);
 
-  constructor(props: any) {
-    super(props);
-  }
+  const handleUpdateDialogSize = React.useCallback(
+    (expanded: boolean): void => {
+      const dialogNode: HTMLDivElement = document.querySelector(
+        '.' + JP_DIALOG_CONTENT
+      );
+      const width = dialogNode.clientWidth;
+      const height = dialogNode.clientHeight;
 
-  updateDialogSize(expanded: boolean): void {
-    if (!this.dialogNode) {
-      this.dialogNode = document.querySelector('.' + JP_DIALOG_CONTENT);
-    }
+      if (
+        expanded &&
+        (width < ERROR_DIALOG_WIDTH || height < ERROR_DIALOG_HEIGHT)
+      ) {
+        setCollapsedSize({ width, height });
+        dialogNode.style.width = Math.max(width, ERROR_DIALOG_WIDTH) + 'px';
+        dialogNode.style.height = Math.max(height, ERROR_DIALOG_HEIGHT) + 'px';
+      } else if (!expanded && collapsedSize) {
+        dialogNode.style.width = collapsedSize.width + 'px';
+        dialogNode.style.height = collapsedSize.height + 'px';
+      }
+    },
+    [collapsedSize, setCollapsedSize]
+  );
 
-    const width = this.dialogNode.clientWidth;
-    const height = this.dialogNode.clientHeight;
+  const details = traceback ? (
+    <ExpandableComponent
+      displayName={'Error details: '}
+      tooltip={'Error stack trace'}
+      onBeforeExpand={handleUpdateDialogSize}
+    >
+      <pre>{traceback}</pre>
+    </ExpandableComponent>
+  ) : null;
```

Comment:
I would normally pull this out into a separate component or include it inline in the jsx below.

since these are string literals I would change them to this:
```
<ExpandableComponent
      displayName="Error details: "
      tooltip="Error stack trace"
      onBeforeExpand={handleUpdateDialogSize}
>
```